java/common/exception/ErrorMessage.java

```diff
@@ -67,14 +67,64 @@
             new ErrorMessage(23, "Rule '%s' 'then' '%s': is trying to assign both an attribute type and a variable attribute value.");
     public static final ErrorMessage INVALID_RULE_THEN_VARIABLES =
             new ErrorMessage(24, "Rule '%s' 'then' variables must be present in rule 'when'.");
-    public static final ErrorMessage REDUNDANT_NESTED_NEGATION =
+    public static final ErrorMessage INVALID_REDUNDANT_NESTED_NEGATION =
```

Comment:
this named seemed better for me since it is more consistent with `INVALID_RULE_WHEN_NESTED_NEGATION` and naming overall (we don't have `REDUNDANT` errors)

Reply:
renamed back